app/dcr_api/biz/service/job_service.go

```diff
@@ -146,3 +211,70 @@ func (js *JobService) GetJobAttestationReport(req *job.QueryJobAttestationReques
 	}
 	return j.AttestationReport, nil
 }
+
+func (js *JobService) getJobOutputFilename(UUID string, originName string) string {
+	return fmt.Sprintf("out-%s-%s", UUID, originName)
+}
+
+func (js *JobService) getJobOutputPath(creator string, UUID string, originName string) string {
+	return fmt.Sprintf("%s/output/%s", creator, js.getJobOutputFilename(UUID, originName))
+}
+
+func (g *JobService) getFileSize(remotePath string) (int64, error) {
+	client, err := storage.NewClient(g.ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to create gcp storage client")
+	}
+	defer client.Close()
+	bucket := config.GetBucket()
+	attr, err := client.Bucket(bucket).Object(remotePath).Attrs(g.ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to get file attributes, or it doesn't exist")
+	}
+	return attr.Size, nil
+}
+
+func (g *JobService) getFilebyChunk(remotePath string, offset int64, chunkSize int64) ([]byte, error) {
```

Comment:
I think all the file operations should completed through an interface such as `StorageBucket` in your refactor doc.